Classes/Model/Download.cs

```diff
@@ -696,25 +697,30 @@ public static int Compare(int epid1, int epid2)
                     switch (sortBy)
                     {
                         case DownloadCols.EpisodeName:
-                            orderBy = "episodes.name" + (sortAsc ? string.Empty : " desc");
+                            orderBy = " order by episodes.name" + (sortAsc ? string.Empty : " desc");
                             break;
                         case DownloadCols.EpisodeDate:
-                            orderBy = "date" + (sortAsc ? string.Empty : " desc");
+                            orderBy = " order by date" + (sortAsc ? string.Empty : " desc");
                             break;
                         case DownloadCols.Status:
-                            orderBy = "status = 0" + (sortAsc ? " desc" : string.Empty) + ", status" + (sortAsc ? " desc" : string.Empty) + ", playcount > 0" + (sortAsc ? string.Empty : " desc") + ", date" + (sortAsc ? " desc" : string.Empty);
+                            orderBy = " order by status = 0" + (sortAsc ? " desc" : string.Empty) + ", status" + (sortAsc ? " desc" : string.Empty) + ", playcount > 0" + (sortAsc ? string.Empty : " desc") + ", date" + (sortAsc ? " desc" : string.Empty);
                             break;
                         case DownloadCols.Duration:
-                            orderBy = "duration" + (sortAsc ? string.Empty : " desc");
+                            orderBy = " order by duration" + (sortAsc ? string.Empty : " desc");
                             break;
                         case DownloadCols.ProgrammeName:
-                            orderBy = "programmes.name" + (sortAsc ? string.Empty : " desc");
+                            orderBy = " order by programmes.name" + (sortAsc ? string.Empty : " desc");
+                            break;
+                        case DownloadCols.SmartName:
+                            // REVIEW - as there is no 'SmartName' field in db to sort on
+                            // sort on ProgrammeName, EpisodeName instead... should produce similar result
+                            orderBy = " order by programmes.name" + (sortAsc ? string.Empty : " desc") + ", episodes.name" + (sortAsc ? string.Empty : " desc");
                             break;
                         default:
                             throw new InvalidDataException("Invalid column: " + sortBy.ToString());
                     }
 
-                    using (SQLiteCommand command = new SQLiteCommand("select downloads.epid from downloads, episodes, programmes where downloads.epid=episodes.epid and episodes.progid=programmes.progid order by " + orderBy, FetchDbConn()))
+                    using (SQLiteCommand command = new SQLiteCommand("select downloads.epid from downloads, episodes, programmes where downloads.epid=episodes.epid and episodes.progid=programmes.progid" + orderBy, FetchDbConn()))
```

Comment:
I'm not sure why you moved the order by from the main SQL statement to the individual case statements..?

Reply:
Made sense at the time when working through a few  options for sorting the 'SmartName' column.
Your call, happy to move it back to the main sql statement.

Reply:
Yes, would prefer it moved back.